packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/sendTransaction.ts

```diff
@@ -1108,6 +1108,8 @@ describe("Eth module", function () {
             );
           }
 
+          const chainId = await this.provider.send("eth_chainId");
```

Comment:
@fvictorio & @alcuadrado, the test was using the `chainId` specified in the provider config, but for a forked provider, we set it based on the remote's chain ID.

I just wanted to validate whether the fix follows the spirit of the original test?